### ISSUE TITLE
Background thread publishing changes when using `RouteSteps`

### ISSUE THREAD

**Issue by utahwithak** [johnpatrickmorgan/FlowStacks#27]
This issue doesn't appear to happen when using the binding modifiers. I'm not sure why. But when using:
``` swift
        RouteSteps.withDelaysIfUnsupported(self, \.routes) {
            $0.goBackToRoot()
        }
```
if there are more than one steps it will print out the following:
```
FlowStacksApp[55359:4275326] [SwiftUI] Publishing changes from background threads is not allowed; make sure to publish values from the main thread (via operators like receive(on:)) on model updates.
```

It appears that inside `scheduleRemainingSteps` when it does a `Task.sleep` when it comes back it will no longer be on the main thread and all remaining operations will execute on a random background thread pool thread. 

Marking `scheduleRemainingSteps` as `@MainActor` solves this issue, though may have other implications. 

This was tested on iOS 15 and 15.5 on simulators. Same thread changing behavior can be seen with a playground as well,

**Comment by DavidKmn** [comment]
PR created #28

**Comment by johnpatrickmorgan** [comment]
Thanks @utahwithak for raising this issue and @DavidKmn for the quick fix!

**Comment by johnpatrickmorgan** [comment]
Merged and released in [V0.2.1](https://github.com/johnpatrickmorgan/FlowStacks/releases/tag/0.2.1).
